public/app/features/auth-config/ProviderConfigForm.tsx

```diff
@@ -74,128 +74,75 @@ export const ProviderConfigForm = ({ config, provider, isLoading }: ProviderConf
+                return (
+                  <CollapsableSection label={section.name} isOpen={index === 0} key={section.name}>
+                    {section.fields.map((field) => {
+                      return (
+                        <FieldRenderer
+                          key={typeof field === 'string' ? field : field.name}
+                          field={field}
+                          control={control}
+                          errors={errors}
+                          setValue={setValue}
+                          register={register}
+                          watch={watch}
+                          unregister={unregister}
+                          secretConfigured={!!config?.settings.clientSecret}
+                        />
+                      );
+                    })}
+                  </CollapsableSection>
+                );
+              })}
+            </Stack>
+          ) : (
+            <>
+              {providerFields.map((field) => {
                 return (
-                  <Select
-                    {...fieldProps}
-                    placeholder={fieldData.placeholder}
-                    isMulti={fieldData.multi}
-                    invalid={invalid}
-                    inputId={name}
-                    options={options}
-                    allowCustomValue
-                    onChange={onChange}
-                    onCreateOption={(v) => {
-                      const customValue = { value: v, label: v };
-                      onChange([...options, customValue]);
-                    }}
+                  <FieldRenderer
+                    key={field}
+                    field={field}
+                    control={control}
+                    errors={errors}
+                    setValue={setValue}
+                    register={register}
+                    watch={watch}
+                    unregister={unregister}
+                    secretConfigured={!!config?.settings.clientSecret}
                   />
                 );
-              }}
-            />
-          </Field>
-        );
-      default:
-        throw new Error(`Unknown field type: ${fieldData.type}`);
-    }
-  };
-
-  return (
-    <Page.Contents isLoading={isLoading}>
-      <Stack grow={1} direction={'column'}>
-        <form onSubmit={handleSubmit(onSubmit)} style={{ maxWidth: '600px' }}>
-          <>
-            <FormPrompt
-              // TODO Figure out why isDirty is not working
-              confirmRedirect={!!Object.keys(dirtyFields).length && !dataSubmitted}
-              onDiscard={() => {
-                reset();
-              }}
-            />
-            <Field label="Enabled">
-              <Switch {...register('enabled')} id="enabled" label={'Enabled'} />
+              })}
+            </>
+          )}
+          <Box display={'flex'} gap={2} marginTop={6}>
+            <Field>
+              <Button type={'submit'}>{isSaving ? 'Saving...' : 'Save'}</Button>
```

Comment:
Could you please also add a `Delete`/`Remove` button with a confirmation modal which says something like `After removing these settings Grafana will use the provider configuration from the system (config file/environment variables) if any.`

Reply:
Done. It issues a `DELETE` request with config provider name as an id, not sure if any other ID should be used instead.

Reply:
The parameter for `DELETE` is the same with the one we use for `GET`. Not to be confused with the `name` property from the sso settings.